src/engine/loaders/obj_loader.cpp

```diff
@@ -71,31 +71,140 @@ void ObjLoader::load(MeshFrame *o_result, char *t_filename, float t_scale, u8 t_
+                        fsetpos(file,&start);
+                        if(newerMatches==3){
+                            /* Configuration confirmed. */
+                            fscanf(file, "%d/// %d/// %d///\n",&vertexIndex[0],&vertexIndex[1],&vertexIndex[2]);
+
+                        }else
+                        {
+                            /* Failed, checking configuration V//VN */
+                            newerMatches = fscanf(file, "%d//%d %d//%d %d//%d",x,x,x,x,x,x);
+                            fsetpos(file,&start);
+                            if(newerMatches==6){
+                                /* Configuration confirmed. */
+                                newerMatches = fscanf(file, "%d//%d %d//%d %d//%d",
+                                        &vertexIndex[0], &normalIndex[0],
+                                        &vertexIndex[1], &normalIndex[1],
+                                        &vertexIndex[2], &normalIndex[2]);
+                            }
+                            else{
+                                /*Unknown configuration.*/
+                                PRINT_ERR("Unknown .obj face for .obj file!");
+                                printf("Obj file mentioned: \"%s\"\n",path);
+                            }
+                        }
+                    }
+                    default:
+                    PRINT_ERR("Unknown faces format in .obj file!");
+                    printf(" \"%s\"\n",path);
+                    break;
+
+                }
+/*                if(ferror(file))
+                    PRINT_ERR("Failed to load .obj file \"%s\".");
                 else
                 {
-                    o_result->getMaterial(materialsI).setVertexFace(faceI, vertexIndex[0] - 1);
-                    o_result->getMaterial(materialsI).setVertexFace(faceI + 1, vertexIndex[1] - 1);
-                    o_result->getMaterial(materialsI).setVertexFace(faceI + 2, vertexIndex[2] - 1);
-
-                    o_result->getMaterial(materialsI).setSTFace(faceI, coordIndex[0] - 1);
-                    o_result->getMaterial(materialsI).setSTFace(faceI + 1, coordIndex[1] - 1);
-                    o_result->getMaterial(materialsI).setSTFace(faceI + 2, coordIndex[2] - 1);
-
-                    o_result->getMaterial(materialsI).setNormalFace(faceI, normalIndex[0] - 1);
-                    o_result->getMaterial(materialsI).setNormalFace(faceI + 1, normalIndex[1] - 1);
-                    o_result->getMaterial(materialsI).setNormalFace(faceI + 2, normalIndex[2] - 1);
-                    faceI += 3;
+                    int result=0;
+                    char line [128]={'\0'};
+                    fgets(line,128,file);
+                    
+                    //if(fscanf
+
+                }*/
+                switch(matches)
+                {
+                    case 9:
+                    {
+                        o_result->getMaterial(materialsI).setVertexFace(faceI, vertexIndex[0] - 1);
```

Comment:
Please check if simple `if`/`if else` will be better there. 
Code will look less redundant.
```
If(maches == 9 || maches == ...) 
  setVertexFace()...
```